worlds/marioland2/__init__.py

```diff
@@ -0,0 +1,374 @@
+            "Tree Zone 2 -> Tree Zone 3": lambda state: has_level_progression(state, "Tree Zone Progression", self.player, 2),
+            "Tree Zone 4 -> Tree Zone 5": lambda state: has_level_progression(state, "Tree Zone Progression", self.player, 3),
+            "Macro Zone 1 -> Macro Zone - Secret Course": lambda state: state.has("Macro Zone Secret 1", self.player),
+            "Macro Zone - Secret Course -> Macro Zone 4": lambda state: state.has("Macro Zone Secret 2", self.player),
+            "Macro Zone 1 -> Macro Zone 2": lambda state: has_level_progression(state, "Macro Zone Progression", self.player),
+            "Macro Zone 2 -> Macro Zone 3": lambda state: has_level_progression(state, "Macro Zone Progression", self.player, 2),
+            "Macro Zone 3 -> Macro Zone 4": lambda state: has_level_progression(state, "Macro Zone Progression", self.player, 3),
+            "Macro Zone 4 -> Macro Zone - Secret Course": lambda state: state.has("Macro Zone Secret 2", self.player),
+            "Pumpkin Zone 1 -> Pumpkin Zone 2": lambda state: has_level_progression(state, "Pumpkin Zone Progression", self.player),
+            "Pumpkin Zone 2 -> Pumpkin Zone - Secret Course 1": lambda state: state.has("Pumpkin Zone Secret 1", self.player),
+            "Pumpkin Zone 2 -> Pumpkin Zone 3": lambda state: has_level_progression(state, "Pumpkin Zone Progression", self.player, 2),
+            "Pumpkin Zone 3 -> Pumpkin Zone - Secret Course 2": lambda state: state.has("Pumpkin Zone Secret 2", self.player),
+            "Pumpkin Zone 3 -> Pumpkin Zone 4": lambda state: has_level_progression(state, "Pumpkin Zone Progression", self.player, 3),
+            "Mario Zone 1 -> Mario Zone 2": lambda state: has_level_progression(state, "Mario Zone Progression", self.player),
+            "Mario Zone 2 -> Mario Zone 3": lambda state: has_level_progression(state, "Mario Zone Progression", self.player, 2),
+            "Mario Zone 3 -> Mario Zone 4": lambda state: has_level_progression(state, "Mario Zone Progression", self.player, 3),
+            "Turtle Zone 1 -> Turtle Zone 2": lambda state: has_level_progression(state, "Turtle Zone Progression", self.player),
+            "Turtle Zone 2 -> Turtle Zone - Secret Course": lambda state: state.has("Turtle Zone Secret", self.player),
+            "Turtle Zone 2 -> Turtle Zone 3": lambda state: has_level_progression(state, "Turtle Zone Progression", self.player, 2),
+            "Menu -> Mario's Castle": lambda state: ([
+                    state.has("Tree Coin", self.player), state.has("Space Coin", self.player),
+                    state.has("Macro Coin", self.player), state.has("Pumpkin Coin", self.player),
+                    state.has("Mario Coin", self.player), state.has("Turtle Coin", self.player)
+                ].count(True) >= self.options.required_golden_coins)
+        }
+        location_rules = {
```

Comment:
The indentation/spacing/styling of this section is inconsistent and rather difficult to parse. It could be cleaned up quite a bit, if you wanted